File: cookbook/copilot-sdk/dotnet/recipe/persisting-sessions.cs
#:package GitHub.Copilot.SDK@*
#:property PublishAot=false

using GitHub.Copilot.SDK;

await using var client = new CopilotClient();
await client.StartAsync();

// Create session with a memorable ID
var session = await client.CreateSessionAsync(new SessionConfig
{
    SessionId = "user-123-conversation",
    Model = "gpt-5"
});

await session.SendAsync(new MessageOptions { Prompt = "Let's discuss TypeScript generics" });
Console.WriteLine($"Session created: {session.SessionId}");

// Destroy session but keep data on disk
await session.DisposeAsync();
Console.WriteLine("Session destroyed (state persisted)");

// Resume the previous session
var resumed = await client.ResumeSessionAsync("user-123-conversation");
Console.WriteLine($"Resumed: {resumed.SessionId}");

await resumed.SendAsync(new MessageOptions { Prompt = "What were we discussing?" });

// List sessions
var sessions = await client.ListSessionsAsync();
Console.WriteLine("Sessions: " + string.Join(", ", sessions.Select(s => s.SessionId)));

// Delete session permanently
await client.DeleteSessionAsync("user-123-conversation");
Console.WriteLine("Session deleted");

await resumed.DisposeAsync();
await client.StopAsync();

File: cookbook/copilot-sdk/python/recipe/persisting_sessions.py
#!/usr/bin/env python3

from copilot import CopilotClient

client = CopilotClient()
client.start()

# Create session with a memorable ID
session = client.create_session(
    session_id="user-123-conversation",
    model="gpt-5",
)

session.send(prompt="Let's discuss TypeScript generics")
print(f"Session created: {session.session_id}")

# Destroy session but keep data on disk
session.destroy()
print("Session destroyed (state persisted)")

# Resume the previous session
resumed = client.resume_session("user-123-conversation")
print(f"Resumed: {resumed.session_id}")

resumed.send(prompt="What were we discussing?")

# List sessions
sessions = client.list_sessions()
print("Sessions:", [s["sessionId"] for s in sessions])

# Delete session permanently
client.delete_session("user-123-conversation")
print("Session deleted")

resumed.destroy()
client.stop()
